ui/de-lib/src/main/java/org/iplantc/de/apps/client/views/toolBar/AppsViewToolbarImpl.java

```diff
@@ -236,9 +236,15 @@ public void onAppSelectionChanged(final AppSelectionChangedEvent event) {
                 final boolean isOwner =
                         selectedApp.getPermission() != null && selectedApp.getPermission()
                                                                           .equals(PermissionValue.own);
-                final boolean isEditable =
-                        isOwner || selectedApp.getPermission() != null && selectedApp.getPermission()
-                                                                                     .equals(PermissionValue.write);
+                final String integratorEmail = selectedApp.getIntegratorEmail();
+                String email = userInfo.getEmail();
+
+                final boolean isEditable = isOwner || (selectedApp.getPermission() != null && selectedApp
+                        .getPermission()
+                        .equals(PermissionValue.write)) || (selectedApp.isPublic()
+                                                            && integratorEmail.equalsIgnoreCase(email));
+
+
```

Comment:
This logic looks good to me, but maybe we can clean up the formatting a little?
It also might help readability to refactor some of this into separate booleans, so maybe something like
```java
final boolean isEditable = isOwner || hasWritePermission || isIntegrator
```